pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -3506,15 +3505,13 @@ public void onUpdate(TopicPolicies policies) {
             replicators.forEach((name, replicator) -> replicator.updateRateLimiter());
             shadowReplicators.forEach((name, replicator) -> replicator.updateRateLimiter());
             checkMessageExpiry();
-            checkReplicationAndRetryOnFailure();
-
-            checkDeduplicationStatus();
-
-            preCreateSubscriptionForCompactionIfNeeded();
-
-            // update managed ledger config
-            checkPersistencePolicies();
-        }).exceptionally(e -> {
+        })
+        .thenCompose(__ -> checkReplicationAndRetryOnFailure())
+        .thenCompose(__ -> checkDeduplicationStatus())
+        .thenCompose(__ -> preCreateSubscriptionForCompactionIfNeeded())
+        .thenCompose(__ -> checkPersistencePolicies())
```

Comment:
Can this PR fix the second issue in onUpdate?

Reply:
yes, right

Reply:
The second issue said there are exceptions that happened when calling `checkPersistencePolicies`. Can we confirm that?

And is it possible that no exceptions happening on checkPersistencePolicies but the topic is loaded before the topic policy is loaded? And the trim task also happened before the topic policy is loaded.

Reply:
> The second issue said there are exceptions that happened when calling checkPersistencePolicies. Can we confirm that

The user said he didn't notice any warn log.

> And is it possible that no exceptions happening on checkPersistencePolicies but the topic is loaded before the topic policy is loaded? And the trim task also happened before the topic policy is loaded.

yes, it's possible